### DISCUSSION TITLE
Fail to set attribute value

### DISCUSSION THREAD

**Discussion by NickeHa** [preactjs/preact#3368] [Q&A]
We developed a web component which have an attribute named value. 
The problem we face is that preact seems to ignore this attribute when we set a value to it.
It works fine if we rename the attribute in the web component to something else.

Is there some special handling of the "value" attribute in preact?

**Comment by JoviDeCroock** [comment]
I wonder whether we're hitting [this branch](https://github.com/preactjs/preact/blob/master/src/diff/props.js#L121) the only other possibility is [this one](https://github.com/preactjs/preact/blob/master/src/diff/index.js#L430) which eventually comes back to the above line.

We can test this by doing

```js
const webComp = document.createElement('your-web-component');
webComp.value = "x" // does this work?

webComp.setAttribute("value", "x") // does this work?
```

**Reply by NickeHa** [reply]
They both seem to work. So I guess it´s only when the attribute is set when the component is created that the problem occur.
It is currently not a major issue for us since we handled the problem by adding an extra attribute to the web component. 
So it is more of a question if there are any special treatment for the value attribute which I guess you did answer :-)

**Reply by JoviDeCroock** [reply]
Hmm, I'd love to see if we can help out more here but it's quite hard to triangulate as those are the only places where we specifically check that prop.

**Reply by NickeHa** [reply]
Thanks for your help! I am currently occupied with other tasks but I will investigate this further when I got more time!

**Reply by NickeHa** [reply]
I think that the reason why it is working bad in our case is that we have a property with the same name as the attribute and that preact is prioritizing the property. So, we need to check why in this case the property doesn't work as expected in the web component. It might just be a coincidence that the attribute and the property's name are "value".
